dockstore-common/src/main/java/io/dockstore/common/yaml/YamlNotebook.java

```diff
@@ -154,7 +155,7 @@ public void setAuthors(final List<YamlAuthor> authors) {
         this.authors = authors;
     }
 
-    public List<String> getTestParameterFiles() {
+    public List<@AbsolutePath String> getTestParameterFiles() {
```

Comment:
Could add a `@NotNull` here (and in the same place in the other entry-type yaml classes):
```
public List<@NotNull @AbsolutePath String> getTestParameterFiles() {
```

Reply:
I left this one to make null values validated since it is allowed in the cli `[here](https://github.com/dockstore/dockstore-cli/blob/f1bdd583546a31f8c671ebf47fdb4ecc006f6122/dockstore-cli-integration-testing/src/test/java/io/dockstore/client/cli/YamlClientIT.java#L101)`

Reply:
> I left this one to make null values validated since it is allowed in the cli [here](https://github.com/dockstore/dockstore-cli/blob/f1bdd583546a31f8c671ebf47fdb4ecc006f6122/dockstore-cli-integration-testing/src/test/java/io/dockstore/client/cli/YamlClientIT.java#L101)

Please correct me if I'm wrong, this stuff can be pretty confusing, but those tests confirm that we accept an empty list of test parameter files, for example:

```
someProperty: foo
testParameterFiles:
someOtherProperty: foo
```

The annotations in `List<@NotNull @AbsolutePath String>` are checking that each _element_ of the list is an absolute path and not null.  For example, it would flag this construction, which is a list of a single null element (`null` is a keyword in yaml):

```
somePropertry: foo
testParameterFiles:
  - null
someOtherProperty: foo
```

This isn't super important, since it's unlikely to happen accidentally, but it keeps the turkeys from sneaking nulls into our system to make it malfunction...